src/som/vmobjects/SClass.java

```diff
@@ -63,14 +64,29 @@
   @CompilationFinal private ClassFactory instanceClassGroup; // the factory for this object
 
   protected final SObjectWithClass enclosingObject;
+  private MaterializedFrame context;
 
   public SClass(final SObjectWithClass enclosing) {
     this.enclosingObject = enclosing;
+    this.context = null;
   }
 
   public SClass(final SObjectWithClass enclosing, final SClass clazz) {
     super(clazz, clazz.getInstanceFactory());
     this.enclosingObject = enclosing;
+    this.context = null;
+  }
+
+  public void setContext(final MaterializedFrame context) {
+    this.context = context;
+  }
+
+  public boolean hasContext() {
+    return context != null;
+  }
+
+  public MaterializedFrame recallContext() {
```

Comment:
So it's different from the ObjectWithContext's `getContext()`. The subclass `SClass` is the class that receives the frame, but the superclass `SObjectWithClass` implements the `ObjectWithContext` method.